client/components/domain-search-v2/domain-search-filters/index.jsx

```diff
@@ -0,0 +1,206 @@
+import clsx from 'clsx';
+import { localize } from 'i18n-calypso';
+import { includes, isEqual, pick } from 'lodash';
+import PropTypes from 'prop-types';
+import { createRef, Component } from 'react';
+import TokenField from 'calypso/components/token-field';
+import ValidationFieldset from 'calypso/signup/validation-fieldset';
+
+import './style.scss';
+
+const HANDLED_FILTER_KEYS = [ 'tlds', 'exactSldMatchesOnly' ];
+
+export class DropdownFilters extends Component {
+	static propTypes = {
+		availableTlds: PropTypes.array,
+		filters: PropTypes.shape( {
+			exactSldMatchesOnly: PropTypes.bool,
+			tlds: PropTypes.array,
+		} ).isRequired,
+		lastFilters: PropTypes.shape( {
+			exactSldMatchesOnly: PropTypes.bool,
+			tlds: PropTypes.array,
+		} ).isRequired,
+		popoverId: PropTypes.string,
+		showTldFilter: PropTypes.bool,
+		showTldFilterPlaceholder: PropTypes.bool,
+		onChange: PropTypes.func.isRequired,
+		onReset: PropTypes.func.isRequired,
+		onSubmit: PropTypes.func.isRequired,
+	};
+
+	static defaultProps = {
+		popoverId: 'search-filters-dropdown-filters',
+	};
+
+	state = {
+		showPopover: false,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.button = createRef();
+	}
+
+	togglePopover = ( { discardChanges = true } = {} ) => {
+		this.setState(
+			{
+				showPopover: ! this.state.showPopover,
+			},
+			() => {
+				if ( discardChanges && ! this.state.showPopover ) {
+					this.props.onChange( this.props.lastFilters );
+				}
+			}
+		);
+	};
+
+	getFiltercounts() {
+		return (
+			( this.props.lastFilters.tlds?.length || 0 ) +
+			( this.props.lastFilters.exactSldMatchesOnly && 1 )
+		);
+	}
+
+	updateFilterValues = ( name, value ) => {
+		this.props.onChange( {
+			[ name ]: value,
+		} );
+	};
+
+	handleExactMatchCheckboxChange = ( value ) => {
+		this.updateFilterValues( 'exactSldMatchesOnly', value );
+	};
+
+	handleOnChange = ( event ) => {
+		const { currentTarget } = event;
+		if ( currentTarget.type === 'checkbox' ) {
+			this.updateFilterValues( currentTarget.name, currentTarget.checked );
+		} else if ( currentTarget.type === 'number' ) {
+			this.updateFilterValues( currentTarget.name, currentTarget.value );
+		}
+	};
+
+	handleFiltersReset = () => {
+		this.togglePopover( { discardChanges: false } );
+		this.props.onReset( 'tlds', 'exactSldMatchesOnly' );
+	};
+
+	handleFiltersSubmit = () => {
+		this.togglePopover( { discardChanges: false } );
+		this.hasFiltersChanged() && this.props.onSubmit();
+	};
+
+	hasFiltersChanged() {
+		return ! isEqual(
+			pick( this.props.filters, HANDLED_FILTER_KEYS ),
+			pick( this.props.lastFilters, HANDLED_FILTER_KEYS )
+		);
+	}
+
+	render() {
+		const hasFilterValues = this.getFiltercounts() > 0;
+
+		return (
+			<div
+				className={ clsx( 'search-filters__dropdown-filters', {
+					'search-filters__dropdown-filters--has-filter-values': hasFilterValues,
+					'search-filters__dropdown-filters--is-open': this.state.showPopover,
```

Comment:
I think we can remove these classes because they're getting in the way:

<img width="259" height="145" alt="image" src="https://github.com/user-attachments/assets/b78aebb8-7abb-4dae-ad38-9071c100a6b2" />